flink-ml-lib/src/main/java/org/apache/flink/ml/evaluation/binaryclassification/BinaryClassificationEvaluator.java

```diff
@@ -673,25 +510,26 @@ public static class BinarySummary implements Serializable {
         public Integer taskId;
         // maximum score in this partition
         public double maxScore;
-        // real positives in this partition
-        public long curPositive;
-        // real negatives in this partition
-        public long curNegative;
+        // sum of weights of positives in this partition
+        public double sumWeightPos;
+        // sum of weights of negatives in this partition
+        public double sumWeightNeg;
 
         public BinarySummary() {}
 
-        public BinarySummary(Integer taskId, double maxScore, long curPositive, long curNegative) {
+        public BinarySummary(
+                Integer taskId, double maxScore, double sumWeightPos, double sumWeightNeg) {
             this.taskId = taskId;
             this.maxScore = maxScore;
-            this.curPositive = curPositive;
-            this.curNegative = curNegative;
+            this.sumWeightPos = sumWeightPos;
+            this.sumWeightNeg = sumWeightNeg;
         }
     }
 
     /** The evaluation metrics for binary classification. */
     public static class BinaryMetrics {
         /* The count of samples. */
-        public long count;
+        public double count;
```

Comment:
This double variable 'count' is not a count yet. The name here is not suitable. Maybe 'weightSum' or 'accValue' is better.

Reply:
I agree with you. I've made changes to all places in the codes.